### ISSUE TITLE
Document the new controller-id flag in the Troubleshooting guide

### ISSUE THREAD

**Issue by simonpasquier** [prometheus-operator/prometheus-operator#6411]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Now that #6319 merged, we should add instructions to https://prometheus-operator.dev/docs/operator/troubleshooting/#prometheusalertmanager-pods-stuck-in-terminating-loop-with-healthy-start-up-logs explaining how you can use controller-id to prevent shared ownership on objects.

### Prometheus Operator Version

```shell
main
```

### Kubernetes Version

```yaml
N/A
```

### Kubernetes Cluster Type

kind

### How did you deploy Prometheus-Operator?

prometheus-operator/kube-prometheus

### prometheus-operator log output

```shell
N/A
```

**Comment by slashpai** [comment]
@YashPimple Sure go ahead. Generally commenting on issue is enough to start working on 🙂

**Comment by danielmellado** [comment]
@simonpasquier @slashpai this looks somehow stalled, happy to take it if @YashPimple isn't working on it.

**Comment by slashpai** [comment]
@danielmellado sure :)

**Comment by Dharma-09** [comment]
I want to work on this if no one is working.

**Comment by slashpai** [comment]
@Dharma-09 Are you working on this?

**Comment by Dharma-09** [comment]
Yeah, I'm still working on it. I got busy, but I'll open the PR soon.

**Comment by Mberga14** [comment]
@simonpasquier  attempted to add the desired docs in #7461
